sources/api/netdog/systemd-derive/README.md

```diff
@@ -0,0 +1,129 @@
+# systemd-derive
+
+Current version: 0.1.0
+
+
+A macro to serialize structs to `systemd` unit file format
+
+### Description
+
+The `SystemdUnit` and `SystemdUnitSection` macros can be used to serialize structs representing
+`systemd` unit files.
+
+Under the hood, the macros implement `Display` for your structs.  This allows converting the
+structs to a string suitable for writing directly to file.
+
+The implementation is fairly rigid to the way we represent the "INI-like" structure of `systemd` unit
+files. `systemd` differs from standard INI format in that duplicate sections are allowed, along with
+duplicate keys within a section.  The macros expect that you will have a "top-level" struct that
+represents the unit file, with nested structs representing the sections of said file.  These nested
+structs representing sections have fields that are the configuration key/value pairs for that
+section. The expectation is that any validation of fields will happen before serialization; no
+validation happens within the macros.
+
+All struct fields must be either `Option`s or `Vec`s of `String`.  Fields that are `Vec`s will be
+iterated upon and each value of the `Vec` will be serialized.  For structs deriving `SystemdUnit`,
```

Comment:
This limitation is no longer present, right? The type just has to implement `Display` but doesn't need to be a `String`.

Reply:
Good catch - that's right!